opm/autodiff/FlowMain.hpp

```diff
@@ -252,9 +253,67 @@ namespace Opm
 #endif
         }
 
```

Comment:
I think this part should go to a separate file(s) NNCHelpers.hpp(cpp) or something like that. I will still review it here though.